src/components/reusable/datePicker/datepicker.stories.js

```diff
@@ -1,114 +1,217 @@
-      table: {
-        defaultValue: { summary: DATE_PICKER_TYPES.SINGLE },
-      },
-    },
-    size: {
-      options: ['sm', 'md', 'lg'],
+    locale: { control: { type: 'text' } },
+    dateFormat: {
+      options: [
+        'Y-m-d',
+        'm-d-Y',
+        'd-m-Y',
+        'Y-m-d H:i',
+        'Y-m-d H:i:s',
+        'm-d-Y H:i:s',
+        'd-m-Y H:i:s',
+      ],
       control: { type: 'select' },
     },
-    minDate: {
-      control: { type: 'text' },
-    },
-    maxDate: {
-      control: { type: 'text' },
-    },
-    step: {
-      control: { type: 'text' },
+    mode: {
+      options: ['single', 'multiple'],
+      control: { type: 'select' },
     },
+    minDate: { control: { type: 'text' } },
+    maxDate: { control: { type: 'text' } },
   },
 };
 
-const args = {
-  unnamed: 'Date',
-  size: 'md',
-  name: 'datepicker',
-  value: '',
-  datePickerType: 'single',
-  caption: '',
+const disconnectFlatpickr = () => {
+  const calendarElements = document.querySelectorAll('.flatpickr-calendar');
+  calendarElements.forEach((calendar) => calendar.remove());
+};
+
+const InputTemplate = (args) => {
+  // prevents flatpickr calendar overlay from persisting on view change
+  useEffect(() => {
+    return () => {
+      disconnectFlatpickr();
+    };
+  }, []);
+
+  const anchorId =
+    args.nameAttr ||
+    `date-picker-button-${Math.random().toString(36).slice(2, 11)}`;
+
+  const placeholder = getPlaceholder(args.dateFormat);
+
+  return html`<kyn-date-picker
+    .nameAttr="${args.nameAttr}"
+    .locale="${args.locale}"
+    .dateFormat="${args.dateFormat}"
+    ?required="${args.required}"
+    .value="${args.value}"
+    .warnText="${args.warnText}"
+    .invalidText="${args.invalidText}"
+    .altFormat=${args.altFormat}
+    .disable="${args.disable}"
+    .enable="${args.enable}"
+    .mode="${args.mode}"
+    .caption="${args.caption}"
+    ?datePickerDisabled="${args.datePickerDisabled}"
+    ?twentyFourHourFormat="${args.twentyFourHourFormat}"
+    .minDate="${args.minDate}"
+    .maxDate="${args.maxDate}"
+    @on-change=${(e) => action(e.type)(e)}
+  >
+    <label
+      slot="label"
+      class="label-text"
+      for=${anchorId}
+      ?disabled=${args.datePickerDisabled}
+    >
+      ${args.required
+        ? html`<abbr
+            class="required"
+            title=${args._textStrings?.requiredText || 'Required'}
+            aria-label=${args._textStrings?.requiredText || 'Required'}
+            >*</abbr
+          >`
+        : null}
+      ${args.unnamed}
+    </label>
+    <input
+      slot="anchor"
+      type="text"
+      id=${anchorId}
+      name=${args.nameAttr}
+      placeholder=${placeholder}
+      ?disabled=${args.datePickerDisabled}
+      ?required=${args.required}
+      aria-invalid=${args._isInvalid ? 'true' : 'false'}
+    />
+    <span slot="icon" style="line-height: 1;">${unsafeSVG(overflowIcon)}</span>
```

Comment:
for this comment and your 3rd comment about the overflow icon, i was working off of the implementation that they have in the figma file where it would be part of a global filter or something.

i suppose we could just have two options -- input or button, but i originally had it slotted instead of internal for flexibility of the anchor. let me know what you think re: having 2-3 hardcoded internal options, just input, etc.

Reply:
I see that in the old design now. I think what the design is getting at there, is they just want some sort of popup menu to house the datepicker in that scenario. I think that design is not well thought out. The calendar selection should probably still be tied to an actual input, and that's how all of the flatpickr examples work. Even the [inline calendar example](https://flatpickr.js.org/examples/#inline-calendar) still shows an input, though maybe there could be a way to hide it.

I think the most important thing is that everything still works with forms. Slotted elements are also exposed to external styling, since they live in the light dom, something to consider.

Reply:
understood. yeah it's been a little difficult to get the design nailed down on this one. still working to get better about communicating the need to have things more clearly defined. 

in the meantime, i've tried to pack in features that i can rip out when the time comes, which is overkill. but easy to scale it back to more internal, simpler variations.

i'll take another stab with all of this in mind and see if we can get it closer

Reply:
ok this should be a bit more on track. i've set `required` to true in storybook for each component by default, as that's what initializes the validation and it will be easier to test. 

two behaviors that i've noticed with flatpickr:
1) for timepicker, if you click to open the timepicker and do not 'touch it' or modify the time at all, then close it by clicking anywhere outside of the overlay, there will be a moment where it does my standard validation and recognizes that the field is required, that no time has been chosen. however, i think flatpicker then defaults to 12:00

2) on the daterangepicker, when you select only one date, my validations are working as expected. however, when you open the datepicker overlay again, it has reset to no dates selected. this is more of a standard date picker behavior, but just wanted to note that as well.